src/Microsoft.ML.Data/Transforms/TermTransform.cs

```diff
@@ -144,6 +147,41 @@ public sealed class Arguments : ArgumentsBase
             public Column[] Column;
         }
 
+        internal sealed class ColInfo
+        {
+            public readonly string Name;
+            public readonly string Source;
+            public readonly ColumnType TypeSrc;
+
+            public ColInfo(string name, string source, ColumnType type)
+            {
+                Name = name;
+                Source = source;
+                TypeSrc = type;
+            }
+        }
+
+        public class ColumnInfo
```

Comment:
>ColumnInfo [](start = 21, length = 10)

This is a little unfortunate. It may be necessary in the short term, but it seems like we have *two* objects to control the behavior, is it not so?

Reply:
One object (ColumnInfo) used to pass parameters and basically a copy of Arguments.Column and used to pass parameters of how to train transformer, second is just convenient class to store information about schema we work with. 

---
In reply to: [214463662](https://github.com/dotnet/machinelearning/pull/759#discussion_r214463662) [](ancestors = 214463662)

Reply:
Right, I was talking about this being a copy of `Arguments.Column`.

---
In reply to: [214468160](https://github.com/dotnet/machinelearning/pull/759#discussion_r214468160) [](ancestors = 214468160,214463662)

Reply:
It's a weak excuse, but I'm just following what @Pete doing right now. I also don't like duplication, but in same time, Arguments.Column and ColumnInfo is a slightly different, ColumnInfo doesn't have any nullable fields, which is the case for Arguments.Column.

---
In reply to: [214482328](https://github.com/dotnet/machinelearning/pull/759#discussion_r214482328) [](ancestors = 214482328,214468160,214463662)

Reply:
I don't think that's what I'm doing.

Reply:
Yes, the pattern I have been following up until now is, I keep `Arguments.Column` intact, and create a separate set of `ColumnInfo` when needed, to pass to my estimators/transformers. 
The idea is, `ColumnInfo` doesn't have to be mutable and only have fields that can be populated via cmdline.

Ultimately, I guess, we will sweep up all the `Arguments` into the `Microsoft.ML.CommandLine` assembly, together with the corresponding loadable classes and factory methods

---
In reply to: [214484933](https://github.com/dotnet/machinelearning/pull/759#discussion_r214484933) [](ancestors = 214484933,214482328,214468160,214463662)